### DISCUSSION TITLE
Fastify/static decodes(change some bytes) in binary *.ttf file in case of hosting in Deta micros

### DISCUSSION THREAD

**Discussion by suantro** [fastify/fastify#4420] [Q&A] [answered]
I am returning a font file using fastify/static:

```
const app = require('fastify')({
    logger: true
  }); 
  app.register(require('@fastify/static'), {
    root: __dirname 
  })
  app.get('/', async (request, reply) => {
    reply.sendFile('font.ttf',{ cacheControl: false })
  })
  app.listen(3000, (err, address) => {
    if (err) {
      throw err
    }
    console.log( "Server is now listening on " + address);
  })
  module.exports = app; 
```

When hosting on localhost, I get the correct 165 kb source file:

HTTP/1.1 200 OK
accept-ranges: bytes
last-modified: Fri, 15 Jul 2022 11:03:32 GMT
Date: Mon, 14 Nov 2022 07:24:01 GMT
etag: W/"286ac-18201872ba0"
content-type: font/ttf
content-length: 165548
------------------------------ 
content: 00 01 00 00 00 0D 00 80 00 03 00 50 46 46 54 4D 6B BE 47 B9 00 02 ... 

When hosting on Deta micros (deta.sh), I get corrupt 200 kb file:

HTTP/1.1 200 OK
Date: Mon, 14 Nov 2022 07:36:10 GMT
Content-Type: font/ttf
Accept-Ranges: bytes
Etag: W/"286ac-4972121400"
Last-Modified: Mon, 31 Dec 1979 00:00:00 GMT
Server: Deta
------------------------------
content: 00 01 00 00 00 0D 00 EF BF BD 00 03 00 50 46 46 54 4D 6B EF BF BD 47 EF BF BD 00 02 ...

Every byte higher than 0x7F is changed to EF BF BD.
I have no possibility to debug app in deta micros lunux mashine. But if I
clone app back to my comp from deta i get correct 165 kb file. So, 'deta clone'
command return true file, but fastify/static - not. 
Please help with advice where and what to do.

**Comment by suantro** [comment]
[index.zip](https://github.com/fastify/fastify/files/10001646/index.zip)
Files attached

**Reply by climba03003** [reply]
Can you provide a repository contain the code? 
It is not safe for the others to download a `zip` file from internet.

**Reply by suantro** [reply]
https://github.com/SAntropov/F_test  - repository.
https://7gf86e.deta.dev/ - hosted in deta.sh app. It returns file font.ttf using fastify/static

**Reply by suantro** [reply]
Package.json in this repository use + fastify@3.29.3
+ @fastify/static@5.0.2. The reason is Win7 in my comp. But at home comp with win10
the latest versions of Fastify and Static show the same result

**Comment by Uzlopak** [comment]
EF BF BD indicates that something tries to urlencode the file content.

**Reply by suantro** [reply]
Yes, but can i find, is it inside Static(or Send) module or inside Deta micros hosting? The only way i can test hosting is to return file back using "deta" application. It returns true file. May be somthing  is not correct in http-head information, and any proxy encode it? Some information like "encoding" in MIME-type text/json but for binary files?

**Comment by mcollina** [comment]
I think you should contact deta support as they are doing something with your request that they should not be doing.

**Comment by climba03003** [comment, marked as answer]
You are not the only one facing the issue. You should file issue to `deta.sh`.
https://github.com/orgs/deta/discussions/464

**Reply by suantro** [reply]
KaKa, Matteo, thank you! I have wrote to Deta
